Naprawianie bledow

pszczepcio opened the pull request:
Wyjaśnienie dlaczego tyle zmian w jednym request na czacie.
zmiany:
1. usunięcie Transactional z Dao
2. Zmiana w niektórych relacjach cascadeType. ALL na cascadeType.REFRESH
3. zmiana fetch.Eager na LAzy
4. Dodanie Transactional w Testach

pszczepcio committed:
1). OrderDaoTestSuite:
- usunięcie cartDao.deletById() (ta encja jest usuwana wraz z inną encją równoczesnie)
- zmiana w tescie metody TestOrderDaoFindAll() - było podwojne zapisanie do zmiennej order. Należło to rozdzielić na order i order 2
- dodanie zapisu koszyka do bazy dancyh cartDao.save(cart2)
- zmiana Long order2Id = order.get(id) na Long order2Id = order2.get(id)
(do dwóch zmiennych był przypisany taki sam Id z encji order a id order2 nie było do niczego przypisane)

2. CartDaoTestSuite:
- usunięcie podwójnego zapisu listy produktów w koszyku. Najpierw do koszyka był dodawany produkt a później do koszyk była przypisana pusta Arraylista
- usunięcie productDao.deleteById() encja produkt jest automatycznie usuwana z koszykiem

pszczepcio committed:
1. usunięcie Transactional z Dao
2. Zmiana w niektórych relacjach cascadeType. ALL na cascadeType.REFRESH
3. zmiana fetch.Eager na LAzy
4. Dodanie Transactional w Testach

klikuskodilla commented on src/test/java/com/kodilla/ecommercee/CartDaoTestSuite.java:
```diff
@@ -70,7 +70,7 @@ public void saveTest() {
 
         //CleanUp
         cartDao.deleteById(cartId);
```
napisz test który sprawdzi czy cascade = CascadeType.REFRESH działa tak jak powinien przypadku cart i order.
Zbędne komentarze do usunięcia

pszczepcio replied:
Metody testów dodane do CartDaoTestSuite i OrderDaoTestsuite.

pszczepcio committed:
- dodanie metody contractCartAndProductRelationshipTest() testującej  relację Cart i Product w testach CartDaoTestSuite
- dodanie metody  contractOrderAndProductRelationshipTest() testującej  relację Order i Product w testach OrderDaoTestSuite

pszczepcio committed:
- usuniecie zbędnych komentarzy w cartDaoTestSuite
